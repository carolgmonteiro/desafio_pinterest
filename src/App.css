@import url("https://fonts.googleapis.com/css?family=Roboto:400,500,700,900&display=swap");
p,
h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: "Roboto", sans-serif;
}
.my-masonry-grid {
  display: -webkit-box; /* Not needed if autoprefixing */
  display: -ms-flexbox; /* Not needed if autoprefixing */
  display: flex;
  margin-left: -30px;
  /* gutter size offset */
  width: auto;
}
.my-masonry-grid_column {
  padding-left: 30px;
  /* gutter size */
  background-clip: padding-box;
}
.my-masonry-grid_column > div {
  /* change div to reference your elements you put in <Masonry> */
  margin-bottom: 10px;
}
.images-container {
  justify-content: center;
  margin-top: 18vh;
  padding-left: 5%;
  padding-right: 2%;
}
@media (min-width: 360px) and (max-width: 768px) {
  /* NavBar */
  #navbar {
    display: inline-block;
    width: 100%;
    min-height: 65px;

    background: white;
    flex-wrap: wrap;
    z-index: 9999;
  }
  .nav-box {
    /* display: inline-flex; */
    background: white;
    max-height: 250px;
    padding-right: 4%;
  }
  .search-box-content {
    flex-basis: 100%;
    display: flex;
    /* flex-wrap: wrap; */
  }
  .images-container {
    justify-content: center;
    margin-top: 25vh;
    padding-left: 5%;
    padding-right: 2%;
  }
  .logo-pinterest {
    margin-top: 3px;
    margin-left: 15px;
    border: none;
    background: transparent;
  }
  .search-box input {
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
    height: 75%;
    width: 100%;
    background: #efefef;
    border: 0;
    font-weight: 600;
    padding: 0;
    cursor: text;
    min-width: 150px;
  }
  .icon {
    justify-content: center;
    /* padding: 2%; */
    padding: 10px;
    height: 30px;
    margin-right: 2px;
    width: 30px;
    display: flex;
  }
}

/* Optional, different gutter size on mobile */
@media (max-width: 800px) {
  .my-masonry-grid {
    margin-left: -15px; /* gutter size offset */
  }
  .my-masonry-grid_column {
    padding-left: 15px; /* gutter size offset */
  }
  .my-masonry-grid_column > div {
    margin-bottom: 15px; /* space between items */
  }
}

/* NavBar */
#navbar {
  /* display: flex; */
  width: 100%;
  height: 65px;
  background: white;
  flex-wrap: wrap;
  z-index: 9999;
}

.header-content {
  z-index: 9999;
  top: 0;
  position: fixed;
  box-sizing: border-box;
}
.nav-box {
  /* display: inline-flex; */
  background: white;
}

#btn-pinterest img {
  height: 28px;
  width: 28px;
}
.logo-pinterest img:hover {
  background: #efefef;
  border-radius: 50%;
}

.logo-pinterest {
  margin-top: 3px;
  margin-left: 15px;
  border: none;
  background: transparent;
}

.search-box-content {
  flex-basis: 100%;
  display: flex;
}

.button-content {
  /* margin-top: 5px; */
  flex-basis: 25%;
  display: flex;
  align-content: space-between;
}

.border {
  height: 50px;
  width: 2px;
  border-left: 2px solid #efefef;
  box-sizing: border-box;
  margin-left: 16px;
  display: block;
  margin-top: 15px;
}
hr {
  background: #8e8e8e;
  width: 100%;
  border-top: 1px solid #efefef !important;
  display: block;
  z-index: 9999;
  margin-top: 0 !important;
  margin-bottom: 0 !important;
  border-bottom: 0;
  border-left: 0;
  justify-content: flex-end;
}

/* Search box */
.search-box {
  margin-top: 20px;
  flex-basis: 100%;
  display: flex;
}

.btn-search {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  color: #8e8e8e;
  padding-top: 6px;
  padding-left: 10px;
  margin-bottom: 5px;
  margin-left: 10px;
  display: flex;
  flex-basis: 15%;
  background: #efefef;
  font-weight: bold;
  font-size: 16px;
  border: none;
  height: 75%;
  width: 100%;
  cursor: pointer;
}

.search-box img {
  padding: 8px;
  margin-bottom: 15px;
  justify-content: center;
  margin-bottom: 15px;
  width: 38px;
  height: 38px;
  display: flex;
}

.search-box input {
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
  height: 75%;
  width: 100%;
  background: #efefef;
  border: 0;
  font-weight: 600;
  padding: 0;
  cursor: text;
  min-width: 200px;
}

/* Bton NavBar */
.button-box {
  margin-top: 20px;
  display: flex;
  align-content: space-between;
}

.btn-nav-name {
  height: 50px;
  width: 100%;
  color: #8e8e8e;
  /* margin-bottom: 10px; */
  margin-left: 5px;
  display: flex;
  flex-basis: 8%;
  background-color: white;
  font-weight: bold;
  font-size: 16px;
  border-radius: 25px;
  padding-left: 15px;
  padding-right: 15px;
  /* padding-top: 8px; */
  /* padding-bottom: 8px; */
  border: none;
  transition: 0.3s;
}
.btn-nav-name:hover {
  background-color: #f2f2f2;
  cursor: pointer;
  height: 50px;
  width: 100%;
}

.user-icon {
  margin-top: 13px;
  margin-right: 6px;
  height: 24px;
  width: 24px;
  border-radius: 50%;
  background: #8e8e8e;
  color: white;
  box-sizing: border-box;
}

.username {
  margin-top: 13px;
}

/* Btn NavBarIcon */

.btn-nav {
  color: #8e8e8e;
  margin-left: 10px;
  display: flex;
  flex-basis: 8%;
  background: transparent;
  border-radius: 20px;
  border: none;
  cursor: pointer;
}
.btn-nav :hover {
  background: #f2f2f2;
  border-radius: 50%;
}
.btn-nav button {
  outline: none;
}

.icon {
  justify-content: center;
  /* padding: 2%; */
  padding: 10px;
  height: 48px;
  margin-right: 2px;
  width: 48px;
  display: flex;
}

.btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  /* padding: 0.375rem 0.75rem; */
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}

.search-box {
  margin-top: 20px;
  flex-basis: 100%;
  display: flex;
}

.btn-search {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  color: #8e8e8e;
  padding-top: 6px;
  padding-left: 10px;
  margin-bottom: 5px;
  margin-left: 10px;
  display: flex;
  flex-basis: 8%;
  background: #efefef;
  font-weight: bold;
  font-size: 16px;
  border: none;
  height: 75%;
  width: 100%;
  cursor: pointer;
}

.search-box img {
  padding: 8px;
  margin-bottom: 15px;
  justify-content: center;
  margin-bottom: 15px;
  width: 38px;
  height: 38px;
  display: flex;
}

.search-box input {
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
  height: 75%;
  width: 100%;
  background: #efefef;
  border: 0;
  font-weight: 600;
  padding: 0;
  cursor: text;
}
