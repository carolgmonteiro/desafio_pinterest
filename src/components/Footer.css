.footer {
  bottom: 14px;
  position: fixed;
  right: 14px;
  -webkit-transform: translateZ(0);
  transform: translateZ(0);
  z-index: 660;
}

.btn-plus,
.btn-question {
  background-color: white;
  outline: none;
  font-weight: 900;
  border-radius: 50%;
  font-size: 25px;
  color: #333;
  border: none;
  width: 45px;
  height: 45px;
  margin: 8px;
  text-align: center;
  box-shadow: 0 2px 0 0 rgba(0, 0, 0, 0.1), 0 0 0 0.5px rgba(0, 0, 0, 0.04);
}

.btn-plus:hover,
.btn-question:hover {
  background: #efefef;
}
