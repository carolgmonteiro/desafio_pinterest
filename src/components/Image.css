.image-container {
  width: 220px;
  padding-bottom: 42px;
  padding-top: 8px;
  padding-left: 8px;
  padding-right: 8px;
  border-radius: 10px;
  position: relative;
  justify-content: center;
}
.image-container:hover {
  background: #efefef;
}

.linked-img {
  border-radius: 10px;
  width: 100%;
}

.linked-img:hover {
  cursor: zoom-in;
  filter: brightness(0.7);
}

.btn-save-collection {
  display: flex;
  flex-wrap: wrap;
}

.btn-save-collection button {
  height: 40px;
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
  font-weight: bold;
  width: 72px;
  border: none;
  color: white;
  background: #e60023;
  position: relative;
  justify-content: center;
  outline: none;
}

.btn-save-collection button:hover {
  background: #cf001f;
}

.btn-save-collection select {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  height: 40px;
  width: 128px;
  background: #efefef;
  border: 0;
  font-weight: 600;
  padding: 0;
  cursor: pointer;
  outline: none;
  scrollbar-arrow-color: #333333;
  border-color: transparent transparent rgba(0, 0, 0, 0.1) transparent;
}
.btn-save-collection select :hover {
  background: #e2e2e2;
}
.btn-dots {
  position: absolute;
  height: 20px;
  width: 20px;
  right: 15px;
}

.btn-dots button {
  border: none;
  background: transparent;
  height: 24px;
  outline: none;
  border-radius: 50%;
}
.btn-dots img:hover {
  background: #e4e3e3;
  border-radius: 50%;
}

.btn-dots img {
  right: 10px;
  bottom: 8px;
  cursor: pointer;
  height: 25px;
  width: 25px;
  border-radius: 50%;
  padding: 5px;
}
.link-content {
  display: flex;
  flex-wrap: wrap;
  /* position: absolute;
  bottom: 50px; */
}
.btn-share {
  border-radius: 50%;
  height: 30px;
  width: 30px;
  background: #474747;
  justify-content: center;
  right: 15px;
  cursor: pointer;
  position: absolute;
  bottom: 50px;
}
.btn-share img {
  width: 100%;
}

.btn-share:hover {
  background: #333333;
}

.link-image a {
  padding: 5px;
  border-radius: 20px;
  color: white;
  height: 30px;
  font-weight: bold;
  background: #474747;
  justify-content: center;
  right: 100px;
  cursor: pointer;
  border: none;
  font-size: 14px;
  position: absolute;
  bottom: 48px;
}

.link-image button:hover {
  background: #333333;
}
