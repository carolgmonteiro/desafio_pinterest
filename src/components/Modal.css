@media (min-width: 360px) and (max-width: 768px) {
  .modal-body img {
    max-width: 50%;
    border-bottom-left-radius: 10px;
    border-top-left-radius: 10px;
  }

  .modal-content {
    margin-top: 110px;
    /* display: flex; */
    /* position: relative; */
    border-radius: 10px;
    border: none;
    width: 100%;
    height: 100%;
    background: white;
    -webkit-box-shadow: 1px 0px 12px 1px rgba(194, 194, 194, 1);
    -moz-box-shadow: 1px 0px 12px 1px rgba(194, 194, 194, 1);
    box-shadow: 1px 0px 12px 1px rgba(194, 194, 194, 1);
  }
}

.modal-content {
  margin-top: 110px;
  /* display: flex; */
  /* position: relative; */
  border-radius: 40px;
  border: none;
  width: 100%;
  height: 100%;
  background: white;
  -webkit-box-shadow: 1px 0px 12px 1px rgba(194, 194, 194, 1);
  -moz-box-shadow: 1px 0px 12px 1px rgba(194, 194, 194, 1);
  box-shadow: 1px 0px 12px 1px rgba(194, 194, 194, 1);
}

.modal-header {
  border-bottom: none;
  display: flex;
  justify-content: flex-end;
  padding: 1rem 1rem;
}

.arrow {
  cursor: pointer;
  position: absolute;
  top: 2vw;
  left: -5vw;
  z-index: 9999;
  margin-left: 2%;
}
.arrow img {
  width: 5%;
  width: 43px;
  padding: 10px;
  z-index: 9999;
}
.arrow img:hover {
  background: #e4e3e3;
  border-radius: 50%;
}

.modal.fade.show {
  background: white;
}
.modal-body {
  padding-top: 0px;
  padding: 0px;
  display: flex;
  /* flex-wrap: wrap; */
  /* position: relative;
  flex: 1 1 auto; */
}
.modal-body img {
  max-width: 50%;
  border-bottom-left-radius: 40px;
  border-top-left-radius: 40px;
}

.img-fluid:hover {
  filter: brightness(0.7);
  cursor: pointer;
}

.btn-save-collection-modal {
  display: flex;
  margin-top: 20px;
  height: 50px;
  padding: 0px 24px;
}

.btn-save-collection-modal button {
  height: 40px;
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
  font-weight: bold;
  width: 72px;
  border: none;
  color: white;
  background: #e60023;
  position: relative;
  justify-content: center;
  outline: none;
  /* margin-left: 30px; */
}

.btn-save-collection-modal button:hover {
  background: #cf001f;
}

.btn-save-collection-modal select {
  padding: 0px 14px;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  height: 40px;
  width: 148px;
  background: #efefef;
  border: 0;
  font-weight: 600;
  cursor: pointer;
  outline: none;
  scrollbar-arrow-color: #333333;
  border-color: transparent transparent rgba(0, 0, 0, 0.1) transparent;
}

.btn-save-collection-modal select:after {
  content: "\f078";
  -webkit-appearance: none;
  /* font: normal normal normal 17px/1 FontAwesome;
  color: #0ebeff;
  right: 11px;
  top: 6px;
  height: 34px;
  padding: 15px 0px 0px 8px;
  border-left: 1px solid #0ebeff; */
  position: absolute;
  pointer-events: none;
}

.btn-save-collection-modal select:hover {
  background: #e2e2e2;
}
.btn-icons-modal {
  margin-top: 20px;
  margin-right: 15px;
  margin-left: 30px;
  height: 50px;
}

.btn-icons-modal button {
  border: none;
  background: transparent;
  height: 24px;
  width: 24px;
  outline: none;
  /* border-radius: 50%; */
}
.btn-icons-modal img:hover {
  background: #e4e3e3;
  border-radius: 50%;
}

.btn-icons-modal img {
  cursor: pointer;
  /* height: 40px; */
  width: 40px;
  /* width: 25px; */
  /* border-radius: 50%; */
  padding: 5px;
}

.btn-share-modal {
  /* height: 60px; */
  width: 60px;
  /* background: #474747; */
  justify-content: center;
  /* right: 20px; */
  /* margin-right: 60px; */
  cursor: pointer;
}
.btn-share-modal img {
  width: 100%;
}

.btn-share-modal img:hover {
  background: #e4e3e3;
  border-radius: 50%;
}

.tags {
  padding: 5px;
  color: #333333;
  font-weight: bold;
  justify-content: center;
  right: 60px;
  cursor: pointer;
  border: none;
  font-size: 14px;
  /* position: absolute; */
  bottom: 48px;
  padding: 10px;
  margin-left: 30px;
}

.link-image-modal a {
  padding: 5px;
  border-radius: 20px;
  color: #333333;
  height: 40px;
  font-weight: bold;
  background: #efefef;
  justify-content: center;
  right: 60px;
  cursor: pointer;
  border: none;
  font-size: 14px;
  /* position: absolute; */
  bottom: 48px;
  padding: 10px;
  margin-left: 30px;
  outline: none;
}

.link-image-modal button:hover {
  background: #e2e2e2;
}
.link-image-modal img {
  height: 25px;
  margin-right: 10px;
}

.modal-infos {
  /* border-bottom: none; */
  display: flex;
  flex-wrap: wrap;
  /* width: 900px; */
  /* align-items: flex-start; */
  /* justify-content: flex-end; */
  /* padding: 1rem 1rem; */
  /* border-top-left-radius: 20px;
  border-top-right-radius: 20px; */
  /* flex-direction: row; */
}
