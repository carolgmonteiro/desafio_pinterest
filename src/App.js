import React from "react";
import Navbar from "../src/components/Navbar";
import "./App.css";
import InfiniteScroll from "react-infinite-scroll-component";
import Masonry from "react-masonry-css";

function App() {
  return (
    <main className="App">
      <header>
        <Navbar />
      </header>
      <div className="images-container">
        <p>Hola mundo</p>
      </div>
    </main>
  );
}

export default App;
