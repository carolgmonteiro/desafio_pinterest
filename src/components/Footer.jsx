import React from "react";
import "./Footer.css";

class Footer extends React.Component {
  render() {
    return (
      <section className="footer">
        <div>
          <button className="btn-plus">
            <i>+</i>
          </button>
        </div>
        <div>
          <button className="btn-question">
            <i>?</i>
          </button>
        </div>
      </section>
    );
  }
}

export default Footer;
