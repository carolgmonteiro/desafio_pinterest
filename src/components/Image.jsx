import React from "react";
import "./Image.css";

class Image extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hover: false
    };
  }

  handleHover() {
    this.setState({ hover: !this.state.hover });
  }

  render() {
    var imgStyle;
    var otherImgStyle;
    if (this.state.hover) {
      imgStyle = {
        display: "flex",
        position: "absolute",
        top: "10px",
        right: "10px"
      };
      otherImgStyle = { display: "block" };
    } else {
      imgStyle = { display: "none" };
      otherImgStyle = { display: "none" };
    }

    return (
      <React.Fragment>
        <div
          onMouseEnter={() => this.handleHover()}
          onMouseLeave={() => this.handleHover()}
          className="image-container"
        >
          <img
            className="linked-img"
            src={this.props.url}
            alt="link"
            data-target={"#modal" + this.props.dataTarget}
            data-toggle="modal"
          />
          <div style={imgStyle} className="btn-save-collection">
            <select name="collections">
              <option>illustration</option>
              <option>watercolor</option>
              <option>photos</option>
              <option>inspiration</option>
            </select>
            <button>Save</button>
          </div>
          <div className="btn-dots">
            <button>
              <img src={require("../img/dots.svg")} alt="dots" />
            </button>
          </div>
          <div className="link-content" style={otherImgStyle}>
            <div className="btn-share">
              <img src={require("../img/share.png")} alt="share" />
            </div>
            <div className="link-image">
              <a href={this.props.pageurl} target={"_blank"}>
                {this.props.pageurl.substr(27, 10)}...
              </a>
            </div>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

export default Image;
