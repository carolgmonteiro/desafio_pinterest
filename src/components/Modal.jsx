import React from "react";
import "./Modal.css";

class Modal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hover: false
    };
  }
  // constructor(props) {
  //   super(props);
  //   this.state = {
  //     images: []
  //   };
  //   this.imageURL = this.imageURL.bind(this);
  // }
  // imageURL() {
  //   return this.state.images.map((image, i) => {
  //     return <Modal dataTarget={image.id} key={i} url={image.pageURL} />;
  //   });
  // }
  handleHover() {
    this.setState({ hover: !this.state.hover });
  }

  render() {
    return (
      <React.Fragment>
        <div
          className="modal fade"
          id={"modal" + this.props.id}
          tabIndex="-1"
          role="dialog"
          aria-labelledby="exampleModalLabel"
          aria-hidden="true"
        >
          <div className="modal-dialog modal-lg" role="document">
            <div className="modal-content">
              <div className="arrow" data-dismiss="modal" aria-label="Close">
                <img src={require("../img/back.svg")} alt="btn-back" />
              </div>
              <div className="modal-body">
                <img className="img-fluid" src={this.props.url} alt="linked" />
                <div className="modal-infos">
                  <div className="btn-icons-modal">
                    <img src={require("../img/share-black.PNG")} alt="share" />
                    <img src={require("../img/dots-black.PNG")} alt="dots" />
                  </div>
                  <div className="btn-save-collection-modal">
                    <select name="collections">
                      <option>illustration</option>
                      <option>watercolor</option>
                      <option>photos</option>
                      <option>inspiration</option>
                    </select>
                    <button>Save</button>
                  </div>
                  <div className="tags">
                    <p>Tags: {this.props.tags}</p>
                  </div>
                  <div className="link-image-modal">
                    <a href={this.props.pageurl} target={"_blank"}>
                      <img
                        src={require("../img/arrow-diagonal.PNG")}
                        alt="share"
                      />
                      {this.props.pageurl.substr(27, 10)}...
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

export default Modal;
