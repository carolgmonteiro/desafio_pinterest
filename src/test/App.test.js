import React from "react";
import ReactDOM from "react-dom";
import App from "../App";
import Navbar from "../components/Navbar";
import Image from "../components/Image";
import Modal from "../components/Modal";
import Footer from "../components/Footer";
import SearchBox from "../components/SearchBox";
import { BrowserRouter } from "react-router-dom";
import { mount, shallow, render } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

it("should render correctly with no props", () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});

it("should render correctly with no props", () => {
  const component = shallow(<Navbar />);
  expect(component).toMatchSnapshot();
});

it("should render correctly with no props", () => {
  const component = shallow(<Footer />);
  expect(component).toMatchSnapshot();
});

it("should render correctly with no props", () => {
  const component = shallow(<Modal />);
  expect(component).toMatchSnapshot();
});

it("should render correctly with no props", () => {
  const component = shallow(<Image />);
  expect(component).toMatchSnapshot();
});

it("should render correctly with no props", () => {
  const component = shallow(<SearchBox />);
  expect(component).toMatchSnapshot();
});

// it("renders correctly", () => {
//   const tree = renderer.create(<App />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// it("render correctly Navbar component", () => {
//   const component = renderer.create(<Navbar />).toJSON();
//   expect(component).toMatchSnapshot();
// });

// it("renders without crashing", () => {
//   const div = document.createElement("div");
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

// describe("<Image />", () => {
//   let shallowComp;
//   beforeEach(() => {
//     const props = {
//       key: 1,
//       url:
//         "https://pixabay.com/get/52e5d3414e55a414f6da8c7dda79367d1c3fdcec53536c4870287add9248c05cb8_1280.jpg"
//     };

//     shallowComp = shallow(<Image url={props.url} />);
//   });
//   test("should render 3 imgs- link, dots and share", () => {
//     const imgResult = shallowComp.find("img");
//     expect(imgResult).toHaveLength(3);
//   });
// });

// it("renders without crashing", () => {
//   const div = document.createElement("div");
//   ReactDOM.render(
//     <BrowserRouter>
//       <Image />
//     </BrowserRouter>,
//     div
//   );
//   ReactDOM.unmountComponentAtNode(div);
// });

// describe("<Navbar/>", () => {
//   let shallowComp;
//   beforeEach(() => {
//     shallowComp = shallow(<Navbar />);
//   });

//   test("should have 8", () => {
//     const button = shallowComp.find("button");
//     expect(button).toHaveLength(8);
//   });
// });
