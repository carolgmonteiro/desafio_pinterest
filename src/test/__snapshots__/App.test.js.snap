// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
<Fragment>
  <nav
    className="header-content"
    id="navbar"
  >
    <Navbar />
    <hr />
  </nav>
  <div
    className="images-container"
  >
    <InfiniteScroll
      dataLength={0}
      disableBrowserPullToRefresh={true}
      endMessage={
        <p
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <b>
            Yay! You have seen it all
          </b>
        </p>
      }
      hasMore={true}
      loader={
        <h4>
          Loading...
        </h4>
      }
      next={[Function]}
      pullDownToRefreshContent={
        <h3>
          Pull down to refresh
        </h3>
      }
      pullDownToRefreshThreshold={100}
      releaseToRefreshContent={
        <h3>
          Release to refresh
        </h3>
      }
    >
      <Masonry
        breakpointCols={
          Object {
            "1100": 4,
            "500": 1,
            "700": 2,
            "default": 5,
          }
        }
        className="my-masonry-grid"
        columnClassName="my-masonry-grid_column"
      />
    </InfiniteScroll>
  </div>
  <Footer />
</Fragment>
`;

exports[`should render correctly with no props 2`] = `
<div
  className="btn-box"
>
  <button
    className="logo-pinterest"
    id="btn-pinterest"
  >
    <img
      alt="logo-pinterest"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "logo.svg",
        }
      }
    />
  </button>
  <div
    className="search-box-content"
  >
    <SearchBox />
    <div
      className="button-content"
    >
      <div
        className="button-box"
      >
        <button
          className="btn-nav-name"
        >
          Home
        </button>
        <button
          className="btn-nav-name"
        >
          Following
        </button>
        <button
          className="btn-nav-name"
        >
          <p
            className="user-icon"
          >
            C
          </p>
          <p
            className="username"
          >
            Carolina
          </p>
        </button>
      </div>
      <div
        className="border"
      />
      <button
        className="btn-nav"
      >
        <img
          alt="chat-button"
          className="icon"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "chat.svg",
            }
          }
        />
      </button>
      <button
        className="btn-nav"
      >
        <img
          alt="notifications-button"
          className="icon"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "bell.svg",
            }
          }
        />
      </button>
      <button
        className="btn-nav"
      >
        <img
          alt="settings-button"
          className="icon"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "dots.svg",
            }
          }
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`should render correctly with no props 3`] = `
<section
  className="footer"
>
  <div>
    <button
      className="btn-plus"
    >
      <i>
        +
      </i>
    </button>
  </div>
  <div>
    <button
      className="btn-question"
    >
      <i>
        ?
      </i>
    </button>
  </div>
</section>
`;

exports[`should render correctly with no props 4`] = `
<form
  className="search-box"
>
  <button
    className="btn-search"
    onClick={[Function]}
  >
    <img
      alt="search-button"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "search.svg",
        }
      }
    />
  </button>
  <input
    name="searchText"
    onChange={[Function]}
    placeholder="Search"
    type="search"
    value=""
  />
</form>
`;

exports[`should render correctly with no props 5`] = `
<Fragment>
  <div
    className="image-container"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="link"
      className="linked-img"
      data-target="#modalundefined"
      data-toggle="modal"
    />
    <div
      className="btn-save-collection"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <select
        name="collections"
      >
        <option>
          illustration
        </option>
        <option>
          watercolor
        </option>
        <option>
          photos
        </option>
        <option>
          inspiration
        </option>
      </select>
      <button>
        Save
      </button>
    </div>
    <div
      className="btn-dots"
    >
      <button>
        <img
          alt="dots"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "dots.svg",
            }
          }
        />
      </button>
    </div>
    <div
      className="link-content"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="btn-share"
      >
        <img
          alt="share"
          src="share.png"
        />
      </div>
      <div
        className="link-image"
      >
        <a
          target="_blank"
        >
          undefined...
        </a>
      </div>
    </div>
  </div>
</Fragment>
`;
